app.go

```diff
@@ -843,7 +842,18 @@ func (a *Application) Commit() abci.ResponseCommit {
 	}
 	a.childTxRefs = nil
 
-	go func(height int64, blockHeader abci.Header) {
+	go func(height int64, blockHeader abci.Header, committedTxs []CommittedTx) {
+		for _, tx := range committedTxs {
+			if err := a.EventHandler.LegacyEthSubscriptionSet().EmitTxEvent(tx.result.Data, tx.result.Info); err != nil {
+				log.Error("Emit Tx Event error", "err", err)
+			}
+
+			if len(tx.txHash) > 0 {
+				if err := a.EventHandler.EthSubscriptionSet().EmitTxEvent(tx.txHash); err != nil {
+					log.Error("failed to emit tx event to subscribers", "err", err)
+				}
+			}
+		}
```

Comment:
Looks ok at first look, I wonder if the block even should be emitted first though... probably should check ganache.

Reply:
Yeah, checking with ganache, block events are emitted before tx events. Fixed.